### ISSUE TITLE
Insets and subplots should have no initial history

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#1783]
**Description of the problem**

When we begin a map inset, the history from the figure is inherited.  This can give bad side-effects.  Below are two near-identical scripts. Both plot a Mercator map with another Mercator map as inset.  First the proper way, specifying ? as the unknown map width that will fit the inset:

```
gmt begin good
	gmt basemap -JM6i -R0/80/0/50 -B -BWSen -X0.7i -Y0.3i
	gmt inset begin -DjTL+w2i/3i+o0.2i -M.4i -F+p1p
        	gmt basemap -R160/180/30/55 -JM? -Ba10fg10 -BWSne --MAP_FRAME_TYPE=plain
		echo 170 50 | gmt plot -Sc0.1i -Gred
	gmt inset end
gmt end show
```

This script gives the desired plot:

![two](https://user-images.githubusercontent.com/26473567/66620707-325e7000-eb7d-11e9-8678-6b6936b19fc2.png)

Now, if I simply give -JM without the ?, the script is

```
gmt begin bad
	gmt basemap -JM6i -R0/80/0/50 -B -BWSen -X0.7i -Y0.3i
	gmt inset begin -DjTL+w2i/3i+o0.2i -M.4i -F+p1p
        	gmt basemap -R160/180/30/55 -JM -Ba10fg10 -BWSne --MAP_FRAME_TYPE=plain
		echo 170 50 | gmt plot -Sc0.1i -Gred
	gmt inset end
gmt end show
```

which gives the wrong plot:

![one](https://user-images.githubusercontent.com/26473567/66620743-56ba4c80-eb7d-11e9-82b0-aedeb6e97352.png)

The reason is that the figure's -JM6i is copied into the history, hence -JM finds that before it auto-scales to fit the inset.

This exercise brings up two questions:

1. Since the inset (or subplot panel) is a _new_ plot it should not have _any_ history.  Agreed?
2. With a clean history, -JM will give an error (no history for -JM found).  This matches our GMT6 paper syntax. We should not allow projections such as -JM as shorthand for -JM?.  Agreed?

**Comment by PaulWessel** [comment]
Sorry, let the above just apply to inset for now - subplot may have different rules if **-R -J** are set on the subplot begin line.

**Comment by PaulWessel** [comment]
As for subplots, I think this is the path forward:

1. When **gmt subplot begin** is called, and before parsing options, we should reset the history that will have already been loaded from the gmt.history file from the current figure.
2. Should **gmt subplot begin** be given **-R -J** then that information will be recorded in the subplot history file.
3. Individual panels will then either inherit that common history or have no prior history, thus requiring a region (or get one from the data) and a projection (and width/scale given as ?).

Currently, we do not do 1, hence outside history can leak into the subplot settings.

**Comment by PaulWessel** [comment]
Closed on account of #1819.
